hydra-explorer/src/Hydra/Explorer.hs

```diff
@@ -82,26 +84,33 @@ readModelGetHeadIds = readTVarIO
 main :: IO ()
 main = do
   withTracer (Verbose "hydra-explorer") $ \tracer -> do
+    opts <- execParser hydraExplorerOptions
+    let Options
+          { networkId
+          , port
+          , nodeSocket
+          , startChainFrom
+          } = opts
     explorerState <- newTVarIO (mempty :: ExplorerState)
     let getHeads = readModelGetHeadIds explorerState
-    args <- getArgs
+        chainObserverArgs =
+          Options.toArgNodeSocket nodeSocket
+            <> Options.toArgNetworkId networkId
+            <> toArgStartChainFrom startChainFrom
     race
-      -- FIXME: this is going to be problematic on mainnet.
-      ( withArgs (args <> ["--start-chain-from", "0"]) $
+      ( withArgs chainObserverArgs $
           Hydra.ChainObserver.main (observerHandler explorerState)
       )
-      ( traceWith tracer (APIServerStarted (fromIntegral port :: PortNumber))
-          *> Warp.runSettings (settings tracer) (httpApp tracer getHeads)
+      ( traceWith tracer (APIServerStarted port)
+          *> Warp.runSettings (settings tracer port) (httpApp tracer getHeads)
       )
       >>= \case
         Left{} -> error "Something went wrong"
```

Comment:
Could we print the error here to get better insights into what happened? Perhaps also use dedicated exception type instead of `error`.

Reply:
error is type Unit, there is no much information to be printed or type assertion to make.

there must be an idiomatic way on how to do this, will do some research.

Reply:
Neither `Left` or `Right` is expected here. It would mean that either the chain observer stopped or the web server stopped.

Should: use `concurrently_` or to enforce both to run and rethrow exceptions if there are any.

Alternative would be to fail on both `Left` and `Right`, as neither is expected to return or you could also use `race_`.

Reply:
good catch!

going with `race_`